### ISSUE TITLE
Speed up assets:precompile technique 

### ISSUE THREAD

**Issue by matyapiro31** [tootsuite/mastodon#4493]
**I don't know Ruby at all**, so I only suggests.
Makefile uses file time stamp to skip unchanged file to compile.
I think we can do the same thing with Sprockets.
maybe only add a line like below in `config/initializers/assets.rb`

```ruby
Rails.application.config.assets.precompile = %w(LIST_OF_FILES_THAT_CHANGED_TIME_STAMP_AND_FILES DEPEND_ON _THEM)
```

**Comment by nullkal** [comment]
Unfortunately, mastodon uses webpacker for compiling assets. I don't think this method works.

**Comment by matyapiro31** [comment]
>mastodon uses webpacker

Sprockets also installed.
maybe you can do the same with webpack.

**Comment by matyapiro31** [comment]
I searched and found webpacker also uses `config/initializers/assets.rb`
.

**Comment by ykzts** [comment]
Partially resolved with #4505.
